TestModels/RecursiveShape/test/SimpleRecursiveShapeImplTest.dfy

```diff
@@ -0,0 +1,31 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+include "../src/Index.dfy"
+
+module SimpleRecursiveShapeImplTest {
+  import SimpleRecursiveShape
+  import opened SimpleRecursiveShapeTypes
+  import opened Wrappers
+  method{:test} GetRecursiveShape(){
+    var client :- expect SimpleRecursiveShape.SimpleRecursiveShape();
+    TestGetRecursiveShape(client);
+  }
+
+  method TestGetRecursiveShape(client: ISimpleRecursiveShapeClient)
+    requires client.ValidState()
+    modifies client.Modifies
+    ensures client.ValidState()
+  {
+    var myDataMap: MapWithRecursion := map[];
+    myDataMap := myDataMap["key2" := StructureWithRecursion(content := Some(IntegerValue(42)))];
```

Comment:
nit: Do we need self assignment? Can we not just assign to the key?

Reply:
Done in here: https://github.com/smithy-lang/smithy-dafny/pull/608/commits/1987f7d610adca2a285c4ff7e4f3b4b6941da9d2